### ISSUE TITLE
Docker stats hangs on container with network_mode of another container

### ISSUE THREAD

**Issue by tarunlalwani** [docker/docker#22655]
Steps to reproduce:

Use below `docker-compose.yml` file
```yaml
version: '2'
services:
  base_test:
    image: centos:6.7
    command: top -b
  another:
    image: centos:6.7
    command: top -b
    network_mode: service:base_test
```

now `docker stats --no-stream base_test` would work fine. But `docker stats --no-stream another` will get stuck. Below is the output of `docker info`

```
Containers: 63
 Running: 7
 Paused: 0
 Stopped: 56
Images: 67
Server Version: 1.11.1
Storage Driver: devicemapper
 Pool Name: docker-253:0-202302851-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 12.38 GB
 Data Space Total: 107.4 GB
 Data Space Available: 35.28 GB
 Metadata Space Used: 22.58 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.125 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-327.13.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.15 GiB
Name: localhost.localdomain
ID: MJM3:DRYH:WMXV:CE3F:FTJS:JZDV:BQIQ:SUMU:UTAX:MPYR:BT5T:TGGM
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

Please let me know if any more info is needed

**Comment by tarunlalwani** [comment]
from logs i can see below

```
May 11 15:25:36 localhost.localdomain docker[12986]: time="2016-05-11T15:25:36.632602336+05:30" level=error msg="collecting stats for 631d3950038acf3ff7e7e69a78ba455ad996e4b5f8de1209d691bdb6e57cd904: invalid id: "
```

**Comment by thaJeztah** [comment]
Looks like a duplicate of https://github.com/docker/docker/issues/21848, which was fixed by https://github.com/docker/docker/pull/21904 (which is not in a release yet)

**Comment by tarunlalwani** [comment]
Can you please close this once the fix is released? That way it would be easier to test it out

**Comment by thaJeztah** [comment]
If you have a test machine to install docker on, you can install the experimental builds, which are built from master, so should contain that fix; `curl -fsSL https://experimental.docker.com | sh`

**Comment by tarunlalwani** [comment]
Okie, checked and confirmed working fine in `1.12.0-dev`

**Comment by thaJeztah** [comment]
Awesome! thanks for testing. I'll close this issue as it's a duplicate of https://github.com/docker/docker/issues/21848, and will be fixed in the next (1.12) release
